### PR TITLE
[Coding Conventions] Enforce sending self as first arg of instance methods

### PR DESCRIPTION
**Pull request by Yacouby**
https://iguazio.atlassian.net/browse/ML-6626

This PR addresses  the 3rd python coding convention rule of the conventions that we defined:
"Use self as the first parameter in instance methods."

### CODE REVIEW

File: mlrun/model_monitoring/db/stores/sqldb/models/mysql.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ class ApplicationResultTable(Base, ApplicationResultBaseTable):
     )
 
     @declared_attr
-    def endpoint_id(cls):
+    def endpoint_id(self):
```
Comment (jond01) [inline mlrun/model_monitoring/db/stores/sqldb/models/mysql.py:63]: To the defense of the original author, the official SQLAlchemy examples are confusing:
https://docs.sqlalchemy.org/en/20/orm/mapping_api.html#sqlalchemy.orm.declared_attr

But they all use `cls`, and in one correct example, they specify the `@classmethod` decorator.
However, looks like it's supported only from version 2.0, and we use 1.4 :(

In short - there's no good solution here.

File: tests/model_monitoring/test_stores/test_sql.py [hunk 2]
```diff
@@ -47,7 +47,7 @@ def store_connection(tmp_path: Path) -> str:
 
     @pytest.fixture()
     def _mock_random_endpoint(
-        cls,
+        self,
```
Comment (jond01) [inline tests/model_monitoring/test_stores/test_sql.py]: Can be a class method

File: tests/model_monitoring/test_stores/test_sql.py [hunk 3]
```diff
@@ -131,7 +131,7 @@ def new_sql_store(cls, store_connection: str) -> Iterator[SQLStoreBase]:
             assert (len(list_of_endpoints)) == 0
 
     def test_sql_target_list_model_endpoints(
-        cls,
+        self,
```
Comment (jond01) [inline tests/model_monitoring/test_stores/test_sql.py]: Can be a `@staticmethod`. I updated it and the methods below in #5641

File: tests/model_monitoring/test_stores/test_sql.py [hunk 4]
```diff
@@ -163,7 +163,7 @@ def test_sql_target_list_model_endpoints(
         assert len(filtered_list_of_endpoints) == 1
 
     def test_sql_target_patch_endpoint(
-        cls,
+        self,
```
Comment (jond01) [inline tests/model_monitoring/test_stores/test_sql.py]: Can be a `@staticmethod`

File: tests/model_monitoring/test_stores/test_sql.py [hunk 5]
```diff
@@ -192,7 +192,7 @@ def test_sql_target_patch_endpoint(
         assert endpoint_dict["error_count"] == 2
 
     def test_sql_write_application_event(
-        cls,
+        self,
```
Comment (jond01) [inline tests/model_monitoring/test_stores/test_sql.py]: Can be a `@classmethod`

File: mlrun/common/schemas/model_monitoring/model_endpoints.py [hunk 6]
```diff
@@ -104,6 +104,7 @@ def from_flat_dict(cls, endpoint_dict: dict, json_parse_values: list = None):
         )
 
     @validator("monitor_configuration")
+    @classmethod
     def set_name(cls, monitor_configuration):
```
Comment (jond01) [inline mlrun/common/schemas/model_monitoring/model_endpoints.py:108]: Again on another dependency.
The official documentation of the legacy version we use contains this mistake:
https://docs.pydantic.dev/1.10/usage/validators/

I assume it works even in version 1.10 as I see it in other places.

**Review by jond01 — COMMENTED** [review]
Looks good! Please address the comments in `test_sql.py`.

File: tests/model_monitoring/test_stores/test_sql.py (hunk 2 above)
Comment (Yacouby) [reply tests/model_monitoring/test_stores/test_sql.py]: this method uses the decorator: `pytest.fixture`...
`pytest.fixture` and `classmethod ` are not designed to work together.

Comment (jond01) [reply tests/model_monitoring/test_stores/test_sql.py]: It used to not work together, but it is fixed in the current version of pytest that we use.
For example:
```py
import pytest


class Test:
    @classmethod
    @pytest.fixture()
    def my_fixt(cls) -> int:
        return 1

    @staticmethod
    def test(my_fixt: int) -> None:
        assert my_fixt == 1
```

File: tests/model_monitoring/test_stores/test_sql.py (hunk 3 above)
Comment (jond01) [reply tests/model_monitoring/test_stores/test_sql.py]: Why is it marked as resolved without any comment or change?
Anyway, the PR I mentioned above is in dev. You can remove the changes to this file except for the fixture.

File: tests/model_monitoring/test_stores/test_sql.py [hunk 7]
```diff
@@ -50,6 +50,7 @@ def store_connection(tmp_path: Path) -> str:
         return f"sqlite:///{tmp_path / 'test.db'}"
 
     @pytest.fixture()
+    @classmethod
```
Comment (jond01) [inline tests/model_monitoring/test_stores/test_sql.py]: Only this order works:
```suggestion
    @classmethod
    @pytest.fixture()
```

**Review by jond01 — APPROVED** [review]
Awesome 😎

**Pull request merged**